### PR TITLE
Improve documentation and code organization.

### PR DESCRIPTION
**Pull request by BGR360**
I took some time to clarify some things in writing (and with pictures!) that confused me for a long while when I started using `block-mesh`.

Please take a careful read through the module-level docs in `geometry` and check me for factual incorrectness.

### CODE REVIEW

**Review by bonsairobo — APPROVED** [review]

File: src/geometry.rs [hunk 1]
```diff
@@ -1,3 +1,115 @@
...
+//!
+//! ## Orientation
+//!
+//! The **orientation** of the coordinate system determines which way textures
+//! are displayed on cube faces. Most often, the orientation is "Y up", meaning
+//! that textures are displayed with the top of the image toward the +Y axis.
+//!
+//! # `{N, U, V}` Space
+//!
+//! "`{N, U, V}` space" is basically the **face-local coordinate space**
```
Comment (bonsairobo) [inline src/geometry.rs:91]: The proper math jargon is "tangent space".

File: src/geometry.rs [hunk 2]
```diff
@@ -1,3 +1,115 @@
+//! Voxel geometry and coordinate systems.
```
Comment (bonsairobo) [inline src/geometry.rs:1]: You nailed it!

File: src/geometry/face.rs [hunk 3]
```diff
@@ -88,24 +90,32 @@ impl OrientedBlockFace {
...
     /// there for more info.
     ///
-    /// This is just one way of assigning UVs to voxel quads. It assumes that each material has a single tile texture with
-    /// wrapping coordinates, and each voxel face should show the entire texture. It also assumes a particular orientation for
-    /// the texture. This should be sufficient for minecraft-style meshing.
+    /// This is just one way of assigning UVs to voxel quads. It assumes that
+    /// each material has a single tile texture with wrapping coordinates, and
+    /// each voxel face should show the entire texture. It also assumes a
+    /// particular orientation for the texture. This should be sufficient for
+    /// minecraft-style meshing.
```
Comment (bonsairobo) [inline src/geometry/face.rs:115]: > This should be sufficient for minecraft-style meshing.

I wonder has this held true for brine, seeing as it is one of the most complete Minecraft clients I've seen. Not sure if you're still using block-mesh, but I appreciate this contribution.

**Comment by bonsairobo** [comment]
This PR rocks! Thank you.

File: src/geometry.rs (hunk 1 above)
Comment (BGR360) [reply src/geometry.rs:91]: Oh, thanks! That connects some bridges in my brain :)

File: src/geometry/face.rs (hunk 3 above)
Comment (BGR360) [reply src/geometry/face.rs:115]: I actually didn't understand this statement fully. What is the assumption made about texture orientation? is it regarding how the top/bottom faces' textures are oriented?

I currently am not using block-mesh for the chunk meshing, so I'm not positive if the assumption held true. But I think I will use it to generate item meshes (like the diamond pickaxe you hold in your hand, that model needs to be generated from the pixels of the item texture).

**Pull request merged**